pkg/i2gw/providers/gce/gce_extensions.go

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gce
+
+import (
+	"context"
+	"encoding/json"
+
+	gkegatewayv1 "github.com/GoogleCloudPlatform/gke-gateway-api/apis/networking/v1"
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw"
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw/ir"
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw/notifications"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/ingress-gce/pkg/annotations"
+	backendconfigv1 "k8s.io/ingress-gce/pkg/apis/backendconfig/v1"
```

Comment:
All these additional imports may be unavoidable, but I'm worried that it could balloon the overall size of gwctl, especially if/when more organizations follow this example and we end up depending on every Ingress + Gateway controller. Do you have an idea of how much work it would be to avoid these imports? Maybe we could just use unstructured types to begin with? Alternatively, if you can show that this PR is not making a meaningful difference to the size of the gwctl binary, it may be a non-issue.

Reply:
I think we have to import backendconfigv1, otherwise we won't be able to do read backendConfig from clusters(we need `backendconfigv1.AddToScheme`). However, the other import `k8s.io/ingress-gce/pkg/annotations` we can get rid of that. Described in details in https://github.com/kubernetes-sigs/ingress2gateway/pull/185#discussion_r1744703985.

Reply:
just to make sure, you meant ingress2gateway binary and not gwctl, right?

Reply:
> just to make sure, you meant ingress2gateway binary and not gwctl, right?

🤦 haha yes, whoops.

> we won't be able to do read backendConfig from clusters(we need backendconfigv1.AddToScheme)

I thought we were already converting to unstructured here. If we use unstructured client to list these, we wouldn't need to call AddToScheme. (That does get annoying to work with, so if the binary isn't massively bigger, maybe we should just leave this as is, just trying to understand the order of magnitude here).

Reply:
The issue with unstructured client is, we need to provide namespace when we do the list, like
```
client.Resource(...).Namespace(namespace).List(context.TODO(), metav1.ListOptions{})
```

Though theoretically, the backend/frontend config should live in the same namespace as the services so this information isn't impossible to infer, this could make read gets very messy. So my suggestion is we leave it like this.  

I can look into the size difference in the binary with import, but I don't think it would make a big difference since there isn't much content in the v1 backendconfig package.

Reply:
I tried to compare the size of the verison built of main and this PR with `ls -l`. The size of binary from main branch version is 72.78MB, while the size of binary from this branch is 75.02MB, which is around 3% increase in size.